src/App.tsx

```diff
@@ -1,61 +1,127 @@
+import { ChangeEvent, MouseEvent, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [initTodos, setTodos] = useState(todos);
+  const [title, setTitle] = useState('');
+  const [emptyTitle, setEmptyTitle] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [emptyUser, setEmptyUser] = useState(false);
+
+  let count = 15;
+
+  const handleSelect = (event: ChangeEvent<HTMLSelectElement>) => {
+    setEmptyUser(false);
+    setUserId(+event.target.value);
+  };
+
+  const handleTitle = (event: ChangeEvent<HTMLInputElement>) => {
+    setEmptyTitle(false);
+    setTitle(event.target.value);
+  };
+
+  const createTodo = (event: MouseEvent) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setEmptyTitle(true);
+    }
+
+    if (!userId) {
+      setEmptyUser(true);
+    }
+
+    if (!title.trim() || userId === 0) {
+      return false;
+    }
+
+    count += 1;
+
+    const newTodo = {
+      id: count,
+      title,
+      completed: false,
+      userId,
+      user: getUser(userId),
+    };
+
+    const form = document.querySelector('form');
+
+    if (!form) {
+      return false;
+    }
+
+    form.reset();
```

Comment:
```suggestion

```
no need to work with DOM tools and methods
we work with React)
just update state to initial values